Source/cielim/Private/ProtobufActor.cpp

```diff
@@ -1,35 +1,205 @@
 // Fill out your copyright notice in the Description page of Project Settings.
 
 #include "ProtobufActor.h"
-#include <iostream>
-#include <fstream>
-#include <sstream>
-
+#include "KinematicsUtilities.h"
+#include "Engine/World.h"
 
 // Sets default values
 AProtobufActor::AProtobufActor()
 {
-    // Set this actor to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
-    PrimaryActorTick.bCanEverTick = true;
-
+ 	// Set this actor to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
+	PrimaryActorTick.bCanEverTick = true;
+    this->hasCameras = false;
 }
 
 // Called when the game starts or when spawned
 void AProtobufActor::BeginPlay()
 {
     Super::BeginPlay();
-    this->protobufreader = new ProtobufReader("scenarioAsteroidArrival_UnityViz.bin");
-
+    this->protobufreader = new ProtobufReader("simulation_protobuffer.bin");
+    this->vizmessage = this->protobufreader->ReadInputData();
+    // Check if message has cameras
+    if (vizmessage.cameras().size() > 0) {
+        this->hasCameras = true;
+    }
+    this->SpawnCelestialBodies();
+    this->SpawnSpacecraft();
 }
 
 // Called every frame
 void AProtobufActor::Tick(float DeltaTime)
 {
     Super::Tick(DeltaTime); 
+    // Read input
     this->vizmessage = this->protobufreader->ReadInputData();
-    std::string debugstr = this->vizmessage.DebugString();
-    UE_LOG(LogTemp, Warning, TEXT("vizmessage: %hs"), debugstr.c_str());
-    GEngine->AddOnScreenDebugMessage(-1, 30.f, FColor::White, FString::Printf(TEXT("vizmessage: %hs"), debugstr.c_str()));
+    // Update Actor postions and rotatons
+    this->UpdateCelestialBodies();
+    this->UpdateSpacecraft();
 
 }
 
+/**
+ * @brief CreateCelestialBodyRotationQuat(celestialbody) Creates a Celestial Body Rotation Quaternion
+ * 
+ * @param celestialbody A celestial body object
+ * @return FQuat celestial body's Quaternion 
+ */
+FQuat CreateCelestialBodyRotationQuat(const vizProtobufferMessage::VizMessage_CelestialBody& celestialbody) {
+    
+    // Set vectors
```

Comment:
:small_red_triangle_down: You can remove the comments in the function. They don't say anything the code isn't already directly saying.

Reply:
Fixed